observer/monitor.go

```diff
@@ -0,0 +1,59 @@
+package observer
+
+import (
+	"time"
+
+	blog "github.com/letsencrypt/boulder/log"
+	p "github.com/letsencrypt/boulder/observer/probes"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// monitor contains the parsed, normalized, and validated configuration
+// describing a given oberver monitor
+type monitor struct {
+	name   string
+	period time.Duration
+	probe  string
```

Comment:
I don't fully understand the use of the `probe` string throughout this. It's basically a type or a kind, right, taking the values `http` or `dns` right now? Can we give it a more descriptive name?